File: server/php/src/Application/Domain/Payments/PaymentDetails/CurrencyConversion.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class CurrencyConversion
{
    public ?bool $acceptedByUser = null;
    public ?DccProposal $proposal = null;

    public function toArray(): array
    {
        return [
            'acceptedByUser' => $this->acceptedByUser,
            'proposal' => ($this->proposal ?? new DccProposal())->toArray(),
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/currency_conversion.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class CurrencyConversion
          attr_accessor :accepted_by_user, :proposal

          def initialize(accepted_by_user: nil, proposal: nil)
            @accepted_by_user = accepted_by_user
            @proposal = proposal
          end

          def to_h
            {
              'acceptedByUser' => @accepted_by_user,
              'proposal' => @proposal&.to_h
            }
          end
        end
      end
    end
  end
end